deploy/helm/edot-collector/kube-stack/values.yaml

```diff
@@ -0,0 +1,827 @@
+opentelemetry-operator:
+  manager:
+    extraArgs:
+      - --enable-go-instrumentation
+  admissionWebhooks:
+    certManager:
+      enabled: false
+
+autoGenerateCert:
+  enabled: true
+  recreate: true
+
+crds:
+  create: true
+
+defaultCRConfig:
+  image:
+    repository: "docker.elastic.co/beats/elastic-agent"
+    tag: "8.16.0-SNAPSHOT"
```

Comment:
To @pkoutsovasilis' point, I'd either comment this out or put a placeholder here, and assume implicitly that the agent version from the git revision applies. Otherwise we'll need to keep updating it via tooling on every active branch, and it's going to be very annoying. Even in this PR, it should be 8.17.0 - there's already a separate branch for 8.16.

Not sure how much of a problem this will be for Kibana. Maybe they can just fill the stack version in here?

Reply:
> Not sure how much of a problem this will be for Kibana. Maybe they can just fill the stack version in here?

That might be an issue because Kibana does not render this file, instead shows a command that contains this file's URL.  For example:

```
$ helm install ... -f https://raw.githubusercontent.com/elastic/elastic-agent/3d64b6de77e512c3a300f1e1a5484877b9485c02/deploy/helm/edot-collector/kube-stack/values.yaml
```

The idea is to have something to what we do for installing filebeat; a versioned raw file URL that we can reference for each release: https://raw.githubusercontent.com/elastic/beats/8.16/deploy/kubernetes/filebeat-kubernetes.yaml

Reply:
I am totally open to move this file into another location if the team feels it not reside here.

Reply:
It should reside here if we want to ensure it works correctly for each commit to the release branches in this repository. If we want to link to it, maybe adding it to release artifacts would be nicer?

In any case, if this values file is supposed to be usable as-is without any additional settings, then we need to ensure that the automation which bumps agent versions applies to it as well.

Also, I was wrong previously, the version on `main` should be `9.0.0-SNAPSHOT`.

Reply:
Another option to prevent hard-coding the image is to remove the reference in the values file and dynamically populate it using Helm's set option. Tests would be running for each commit image version:  https://github.com/elastic/elastic-agent/pull/5822/commits/43414b0e7f76f8c0f0251fd68737c5b226d1cbe6

Then, we would need to extend Kibana's provided command too:
```
$ helm upgrade --install --namespace opentelemetry-operator-system opentelemetry-kube-stack open-telemetry/opentelemetry-kube-stack --values https://github.com/.../8.16/.../values.yaml --set defaultCRConfig.image.repository=docker.elastic.co/beats/elastic-agent --set defaultCRConfig.image.tag=8.16.0-SNAPSHOT --version 0.3.0
```

Reply:
> Another option to prevent hard-coding the image is to remove the reference in the values file and dynamically populate it using Helm's set option. Tests would be running for each commit image version: [43414b0](https://github.com/elastic/elastic-agent/commit/43414b0e7f76f8c0f0251fd68737c5b226d1cbe6)
> 
> Then, we would need to extend Kibana's provided command too:
> 
> ```
> $ helm upgrade --install --namespace opentelemetry-operator-system opentelemetry-kube-stack open-telemetry/opentelemetry-kube-stack --values https://github.com/.../8.16/.../values.yaml --set defaultCRConfig.image.repository=docker.elastic.co/beats/elastic-agent --set defaultCRConfig.image.tag=8.16.0-SNAPSHOT --version 0.3.0
> ```

I think `defaultCRConfig.image.repository` can stay. For the rest, it would certainly make our lives maintaining this config easier.

Reply:
Chiming in, the renovate is indeed doing the job but, if my understanding is correct, it is "reactive" in the sense that a version of agent it would have to be released and then it would open a PR to update the file, right?!. Can't we use something similar to [this approach](https://github.com/elastic/elastic-agent/blob/main/magefile.go#L3461) to keep the version up to date!?

Reply:
I'm ok with that as well. But then this PR should actually update the automation and verify that it actually works. Then we'd need to carefully backport it to release branches.

Reply:
@swiatekm @pkoutsovasilis Just updated the PR tagging the elastic-agent version to `9.0.0` and updating the Mage `helm:updateAgentVersion` to update this values file as well. Integration tests have been added, do you know if there is any time limit? The Chart deploys a pod Operator which once it is ready will deploy some more pods (daemonset and cluster/deployment). The Operator deployment time is non-deterministic and varies depending on the environment, thus the `require.Eventuallyf` in the tests.